src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2018,6 +2018,30 @@ private void PerformSchedulingActions(IEnumerable<ScheduleResponse> responses)
             }
         }
 
+        internal int CreateRarNode()
+        {
+            string nodeLocation = _buildParameters?.NodeExeLocation ?? BuildEnvironmentHelper.Instance.CurrentMSBuildExePath;
+            if (string.IsNullOrEmpty(nodeLocation))
+            {
+                // Couldn't find a path to MSBuild.exe; can't create a new node.
+                return -1;
```

Comment:
minor: if the return code is meaningful, should this be a constant also accessible to callers?

Reply:
Fixed!